webapp/src/ts/modules/analytics/analytics-target-aggregates-detail.component.ts

```diff
@@ -0,0 +1,95 @@
+import { ChangeDetectorRef, Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { combineLatest, Subscription } from 'rxjs';
+import { Store } from '@ngrx/store';
+import { TranslateService } from '@ngx-translate/core';
+
+import { Selectors } from '@mm-selectors/index';
+import { TargetAggregatesActions } from '@mm-actions/target-aggregates';
+import { TargetAggregatesService } from '@mm-services/target-aggregates.service';
+import { GlobalActions } from '@mm-actions/global';
+
+@Component({
+  selector: 'analytics-target-aggregates-detail',
+  templateUrl: './analytics-target-aggregates-detail.component.html'
+})
+export class AnalyticsTargetAggregatesDetailComponent implements OnInit, OnDestroy {
+  private targetAggregatesActions: TargetAggregatesActions;
+  private globalActions: GlobalActions;
+  subscriptions: Subscription = new Subscription();
+  aggregates = null;
+  selected = null;
+  error = null;
+  paramTargetId = null; // Defaulting value so initially it'll be different from route's params and show content section
+  shouldLoadDetail;
+
+  constructor(
+    private store: Store,
+    private route: ActivatedRoute,
+    private targetAggregatesService: TargetAggregatesService,
+    private translateService: TranslateService,
+    private changeDetectorRef: ChangeDetectorRef
+  ) {
+    this.targetAggregatesActions = new TargetAggregatesActions(store);
+    this.globalActions = new GlobalActions(store);
+  }
+
+  ngOnInit(): void {
+    const subscription = combineLatest(
+      this.route.params,
+      this.store.select(Selectors.getTargetAggregates),
+      this.store.select(Selectors.getSelectedTargetAggregate),
+      this.store.select(Selectors.getTargetAggregatesError),
+    )
+      .subscribe(([
+        params,
+        aggregates,
+        selected,
+        error
+      ]) => {
+        this.selected = selected;
+        this.error = error;
+        this.aggregates = aggregates;
+        this.shouldLoadDetail = this.paramTargetId !== params.id;
```

Comment:
Correct, I need `aggregates` and the current route (react to its updates too).

Reply:
I think this is the cause of your `changeDetection` error :) As far as I can tell, the error appears when a change happens DURING a change detection execution. 

So, you're in this place where: 
1. your route changed earlier 
2. change detection detects that analytics have changed and within this change detection execution, you change other variable (my bet is on `this.shouldLoadDetail = false;` here: https://github.com/medic/cht-core/pull/6740/files#diff-7b291b0215f0e83074ad3f0140794ce8260ac65895e463991c99b12e3c9b404cR70). 

Could you, please, satisfy my curiosity, and remove this statement and see if the change detection error still occurs?

Reply:
My feeling is that changes that we propagate through redux are async and don't trigger this.

Reply:
I had the error before adding `this.shouldLoadDetail` piece. it happens when updating `this.selected` and calling `this.getAggregatesDetail(this.paramTargetId);` in the same subscription. if I only call it during router change then we don't get the error, but this doesn't cover all scenarios. Wrapping this one in timeout in this subscription also helps, but we want to avoid timeouts. 🤔 It's like we need a little "break of time" between value assignments in 'this.selected'.
So, my fix is to separate the subscribers, one for the "selected agregate" and another for the "aggregates list", looks like this fixed the issue, it makes sense since we just want to do something for when one of those changes.

Reply:
I applied the same idea as in tasks but I wasn't able to get rid of the change detection issue yet, still related with `this.selected` assignment on the first load. At the end I left the `detectChanges` in the afterViewInit that way it'll execute only one time 🙁 still better than first version of this code.